gtsfm/frontend/verifier/ransac.py

```diff
@@ -21,68 +23,91 @@
+        use_intrinsics_in_verification: bool = False,
     ) -> Tuple[Optional[Rot3], Optional[Unit3], np.ndarray]:
-        """Estimates the essential matrix and verifies the feature matches.
-
-        Note: this function is preferred when camera intrinsics are known. The
-        feature coordinates are normalized and the essential matrix is directly
-        estimated.
+        """Performs verification of correspondences between two images to recover the relative pose and indices of
+        verified correspondences.
 
         Args:
             keypoints_i1: detected features in image #i1.
             keypoints_i2: detected features in image #i2.
-            match_indices: matches as indices of features from both images, of
-                           shape (N3, 2), where N3 <= min(N1, N2).
+            match_indices: matches as indices of features from both images, of shape (N3, 2), where N3 <= min(N1, N2).
             camera_intrinsics_i1: intrinsics for image #i1.
             camera_intrinsics_i2: intrinsics for image #i2.
+            use_intrinsics_in_verification (optional): Flag to perform keypoint normalization and compute the essential
+                                                       matrix instead of fundamental matrix. This should be preferred
+                                                       when the exact intrinsics are known as opposed to approximating
+                                                       them from exif data. Defaults to False.
 
         Returns:
             Estimated rotation i2Ri1, or None if it cannot be estimated.
             Estimated unit translation i2Ui1, or None if it cannot be estimated.
-            Indices of verified correspondences, of shape (N, 2) with N <= N3.
-                These indices are subset of match_indices.
+            Indices of verified correspondences, of shape (N, 2) with N <= N3. These are subset of match_indices.
         """
         verified_indices = np.array([], dtype=np.uint32)
 
         # check if we don't have the minimum number of points
-        if match_indices.shape[0] < self.min_pts:
+        if match_indices.shape[0] < self.min_pts_e_matrix:
             logger.info("No match indices were provided to the verifier, returning early with None output")
             return None, None, verified_indices
 
         uv_norm_i1 = feature_utils.normalize_coordinates(keypoints_i1.coordinates, camera_intrinsics_i1)
         uv_norm_i2 = feature_utils.normalize_coordinates(keypoints_i2.coordinates, camera_intrinsics_i2)
         K = np.eye(3)
 
-        i2Ei1, inlier_mask = cv2.findEssentialMat(
-            uv_norm_i1[match_indices[:, 0]],
-            uv_norm_i2[match_indices[:, 1]],
-            K,
-            method=cv2.RANSAC,
-            threshold=NORMALIZED_COORD_RANSAC_THRESH,
-            prob=DEFAULT_RANSAC_SUCCESS_PROB,
-        )
+        if use_intrinsics_in_verification:
+            i2Ei1, inlier_mask = cv2.findEssentialMat(
```

Comment:
can we put
```python
        uv_norm_i1 = feature_utils.normalize_coordinates(keypoints_i1.coordinates, camera_intrinsics_i1)
        uv_norm_i2 = feature_utils.normalize_coordinates(keypoints_i2.coordinates, camera_intrinsics_i2)
        K = np.eye(3)
```
inside the `if use_intrinsics_in_verification:` branch, stince not needed, otherwise?

Reply:
resolved

Reply:
done